File: week1-c/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int calculate_coins(int cents, int coin_value);

int main(void)
{
    // prompt the user for change owed
    int change;
    do
    {
        change = get_int("Change owed (in cents)? ");
    }
    while (change < 0);

    // Calculate how many quarters you should give customer
    int quarters = calculate_coins(change, 25);
    // Subtract the value of those quarters from cents
    change -= quarters * 25;

    int dimes = calculate_coins(change, 10);
    change -= dimes * 10;

    int nickels = calculate_coins(change, 5);
    change -= nickels * 5;

    int pennies = calculate_coins(change, 1);
    change -= pennies * 1;

    int sum = (quarters + dimes + nickels + pennies);

    printf("The minimum coins needed is %i\n", sum);
}

int calculate_coins(int cents, int coin_value)
{
    int coins = 0;
    while (cents >= coin_value)
    {
        coins++;
        cents -= coin_value;
    }
    return coins;
}

File: x/week6-python/sentimental-cash/cash.py
from cs50 import get_float


def main():
    # Prompt the user for the change
    while True:
        change = get_float("Change owed: $")
        if change > 0:
            change *= 100
            break

    # Calculate the minimum number of coins
    quarters = calc_coins(change, 25)
    change -= quarters * 25

    dimes = calc_coins(change, 10)
    change -= dimes * 10

    nickels = calc_coins(change, 5)
    change -= nickels * 5

    pennies = calc_coins(change, 1)
    change -= pennies * 1

    # Print it out
    sum = quarters + dimes + nickels + pennies
    print(sum)


def calc_coins(change, coin_value):
    coins = 0
    while change >= coin_value:
        coins += 1
        change -= coin_value
    return coins


main()
